### PR TITLE
I7026 adding check for clientApp type for guide blocks/pages

### LINKED ISSUE #7026: Page not found when clicking the photo, music & video extensions link from Enhance Your Media Experience
Followup for https://github.com/mozilla/addons-frontend/issues/6701

STR:
1. Load AMO dev on android and click the Enhance Your Media Experience guide
2. Scroll down the page and click on  photo, music & video extensions

Expected result:
Page redirects to https://addons-dev.allizom.org/en-US/android/extensions/photos-media/

Actual results:
Page not found

Notes:
This issue is reproducible on AMO dev with FF63(Android 8.0)

page not found

### LINKED ISSUE #7058: Add-on are not compatible with Android in the new guidelines
Followup for https://github.com/mozilla/addons-frontend/issues/6701

STR:
1. Load AMO dev with Android and click Organize your tabs or Enhance your Media Experience guides
2. Observe if all the section display add-ons

Expected result:
All the guides display add-ons with descriptions and an Install button.

Actual result:
Some sections are missing the add-on given as example. 

Notes:
This issue is reproducible on AMO dev with FF63 (Android 8.0)

Guide lines on Android 

android display

This is how it looks like after changing filters (from android to FF and then back to android)

android display2

### PR DESCRIPTION
**Pull request by rebmullin**
fixes #7026 
fixes #7058

### CODE REVIEW

**Comment by rebmullin** [comment]
> sorry but I was super confused by this title..

I am so sorry. I am making an effort to not do this (it's a bad habit) but sometimes it slips:) I will update!


> I guess we should also update the guide pages to return 404 on Android?

right! I will update this too

**Comment by bobsilverberg** [comment]
It looks like @willdurand has already looked at this PR, so I'm going to assign it to him so someone else doesn't duplicate the effort. If I am incorrect about that I'd be happy to review it myself, just let me know.

File: src/amo/components/HomeHeroGuides/styles.scss [hunk 1]
```diff
@@ -4,12 +4,11 @@ $icon-size: 40px;
 
 .HomeHeroGuides-header {
   color: $black;
-  padding: 0 0 12px;
+  padding: 12px 0 24px;
```
Comment (willdurand) [inline src/amo/components/HomeHeroGuides/styles.scss]: Could we reuse the `$padding-page` variable(s)?

File: src/amo/pages/Guides/index.js [hunk 2]
```diff
@@ -35,6 +36,7 @@ type InternalProps = {|
   addons: {
     [guid: string]: AddonType | null,
   },
+  clientApp: ?string,
```
Comment (willdurand) [inline src/amo/pages/Guides/index.js]: This prop should always be defined, no?

Comment (rebmullin) [reply src/amo/pages/Guides/index.js]: This was the only way to make flow happy if I recall...so hopefully this is ok..?

**Pull request merged**